packages/orchestration/src/orchestration-client.ts

```diff
@@ -50,17 +53,57 @@ export class OrchestrationClient {
 
     return new OrchestrationResponse(response);
   }
+
+  async stream(
+    prompt?: Prompt,
+    requestConfig?: CustomRequestConfig,
+    controller = new AbortController()
+  ): Promise<OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>> {
+    const response = new OrchestrationChatCompletionStreamResponse<OrchestrationChatCompletionStreamChunkResponse>();
+    response.stream = (await this.createStream(controller, prompt, requestConfig))
+      ._pipe(OrchestrationChatCompletionStream._processChunk)
+      ._pipe(OrchestrationChatCompletionStream._processFinishReason, response)
+      ._pipe(OrchestrationChatCompletionStream._processTokenUsage, response);
+    return response;
+  }
+
+  private async createStream(
+    controller: AbortController,
+    prompt?: Prompt,
+    requestConfig?: CustomRequestConfig
+  ): Promise<OrchestrationChatCompletionStream<CompletionPostResponseStreaming>> {
+    const body = constructCompletionPostRequest(this.config, prompt, true);
+    const deploymentId = await resolveDeploymentId({
+      scenarioId: 'orchestration',
+      resourceGroup: this.deploymentConfig?.resourceGroup
+    });
+
+    const response = await executeRequest(
+      {
+        url: `/inference/deployments/${deploymentId}/completion`,
+        resourceGroup: this.deploymentConfig?.resourceGroup
+      },
+      body,
+      {
+        ...requestConfig,
+        signal: controller.signal
+      }
+    );
+    return OrchestrationChatCompletionStream._create(response, controller);
```

Comment:
[pp] This part is a bit inconsistent to Azure OpenAI client. I think I tried to extract the common part out from both streaming and non-streaming variants to avoid code duplication like get deployment ID and construct `endpointOptions`.

Reply:
Yeah that's a good idea, I'll create a common method.